### PR TITLE
Job submission uses functions in `create_batch.py`

### PR DESCRIPTION
**Pull request by lily-wittle**
`process_rsmas.py` now calls function in `create_batch.py` instead of `_processSteps.py` to eliminate duplication of job writing and submitting code. Note that `__main__` of `create_batch.py` submits a batch of multiple jobs where `process_rsmas.py` submits only one job. Also replaced `job_submission_params` namespace with passing individual parameters to each function for clarity.

### CODE REVIEW

**Comment by gravelcycles** [comment]
Could you squash your changes before merging next time?

File: create_batch.py [hunk 1]
```diff
@@ -26,12 +26,13 @@ def create_argument_parser():
     group = parser.add_argument_group("Input File", "File/Dataset to display")
     group.add_argument("file", type=str, help="The file to batch create")
     group.add_argument("--template", dest="template", metavar="TEMPLATE FILE", help="The template file with options")
-    group.add_argument("--memory", dest="memory", metavar="MEMORY (KB)",
+    group.add_argument("--memory", dest="memory", default=3600, metavar="MEMORY (KB)",
```
Comment (gravelcycles) [inline create_batch.py:29]: Is this code related to this diff? In the future, please review the changes made and add other changes to a separate PR

File: create_batch.py [hunk 2]
```diff
@@ -60,132 +57,171 @@ def parse_arguments(args):
...
 
     if scheduler == "PBS":
-        job_file_lines.append("\n#PBS -V")
+        job_file_lines.append(prefix + "-V")
 
     return job_file_lines
 
-def write_job_files(job_submission_params, scheduler = os.getenv('JOBSCHEDULER')):
+
+def write_single_job_file(job_name, command_line, work_dir, project_name, scheduler=os.getenv("JOBSCHEDULER"),
```
Comment (gravelcycles) [inline create_batch.py:125]: I'm not too hyped about having a function call as the default value of a keyword argument. Can we default `scheduler` to `None` and assign it to `os.getenv(...)` in the function body?

File: create_batch.py [hunk 3]
```diff
@@ -60,132 +57,171 @@ def parse_arguments(args):
...
+    Iterates through jobs in input file and writes a job file for each job using the specified scheduler.
+    :param batch_file: File containing batch of jobs for which we are creating job files.
+    :param out_dir: Output directory for run files.
+    :param scheduler: Job scheduler to use for running jobs. Defaults based on environment variable JOBSCHEDULER.
+    :param memory: Amount of memory to use. Defaults to 3600 KB.
+    :param walltime: Walltime for the job. Defaults to 4 hours.
+    :param queue: Name of the queue to which the job is to be submitted. Default is set based on the scheduler.
+    :return: List of job file names.
+    """
+    job_files = []
```
Comment (gravelcycles) [inline create_batch.py:165]: Can we move this line down to right before the `for` loop? Makes it a bit easier to follow

**Review by gravelcycles — COMMENTED** [review]
Overall, super clean and looks easy to use. A couple thoughts:

1. You're in between creating a whole module just for job submission and using it for a specific use case. It seems like you made the right compromises, but if other files start to use this job submission function, I would recommend moving the logic into a separate file since even now it is a bit weird that `process_rsmas` is calling `create_batch` instead of `job_submission` or something like that.
2. If you do extend this to full job submission, I would think about handling other scheduler parameters. At the moment, you've restricted the interface which is okay for the moment.

File: create_batch.py (hunk 1 above)
Comment (lily-wittle) [reply create_batch.py:29]: Just changed to match the `outdir` line that was added when I was rebasing from master. Sorry about that, will do in the future.

**Comment by lily-wittle** [comment]
Merged in changes based on comments. Thanks for the thoughts about a job submission module - I think we discussed it a little bit last week, and it would be a lot cleaner if other files do call its functions. Will keep as is for the time being.

**Pull request merged**